Source/DafnyCore/AST/Expressions/Expressions.cs

```diff
@@ -2258,7 +2259,10 @@ public BinaryExpr(IToken tok, BinaryExpr.ResolvedOpcode rop, Expression e0, Expr
           formatter.SetIndentations(ownedTokens[0], formatter.binOpIndent, formatter.binOpIndent, formatter.binOpArgIndent);
         } else {
           var startToken = this.StartToken;
-          var newIndent = formatter.GetNewTokenVisualIndent(startToken, formatter.GetIndentInlineOrAbove(startToken));
+          var newIndent =
+            ownedTokens[0].val == "," ?
```

Comment:
"," in a comprehension is an "&&", except that it should not try to keep a visual indentation between components.